torch/csrc/jit/codegen/cuda/lower_validation.cpp

```diff
@@ -878,6 +884,81 @@ void validateMmaTensors(MmaOp* mma) {
   validate_operand_ids(mma->inB()->as<TensorView>());
 }
 
+//! Note and TODO:
+//!   Currently relying on ldmatrix to
+//!     obtain the correct data layout for turing/ampere
+//!     mma's.
+//!   This restriction will eventually not
+//!    be necessary once the scatter swizzle is ready.
+void validateTuringInput(TensorView* tv) {
```

Comment:
Pattern matching here to make sure LDMatrix is the right format.
Format is done through swizzling in the scheduling and we check that swizzling to make sure it's correctly setup for LDMatrix.
We could in theory support patterns LDMatrix doesn't support, but that would also mean the MMA isn't supported and so we would have to lower to something completely different.

MemCpy async is a more generic utility that we can use.

Reply:
Updated. Thanks.